### PR TITLE
Add media playlist example

### PR DESCRIPTION
**Pull request by heff**
[Updated media-playlist](https://github.com/muxinc/media-playlist/pull/1) based on [the playlist discussion](https://github.com/muxinc/media-chrome/discussions/97). Adding an example to media-chrome.

### CODE REVIEW

**Review by cjpillsbury — APPROVED** [review]
Since this doesn't impact any fundamental architecture of media chrome, I don't see any reason not to merge this.

**Comment by luwes** [comment]
bug? on my end the first clip that is played doesn't update media chrome state like playing and progress

SCR-20220613-d4t

**Comment by heff** [comment]
> on my end the first clip that is played doesn't update media chrome state
Interesting. Can I see your test? It's working for me at least.
I did have issues with `<youtube-video>`, but I wasn't sure if that was a media-playlist issue or related to the youtube player load timing.

> ideally media-playlist-item's should also support <source> and <track> elements but could be supported later.

[They do!](https://github.com/muxinc/media-playlist/pull/1/files#diff-e727e4bdf3657fd1d798edcd6b099d6e092f8573cba266154583a746bba0f346R248)

**Comment by luwes** [comment]
sweet! sorry, didn't check the code source.

Mac / Chrome 102
https://media-chrome-git-media-playlist-example-mux.vercel.app/examples/media-elements/media-playlist.html
https://stream.new/v/Y576wlRA3xRgjz63Lpr00isKn16OKdBWuHicD4fTrdFg

File: examples/media-elements/media-playlist.html [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      }
+    </style>
+  </head>
+  <body>
+    <main>
+      <h1>Media Chrome Media Playlist Example</h1>
+      <media-controller>
+        <media-playlist slot="media" loop>
+          <media-playlist-item type="hls-video" crossorigin src="https://stream.mux.com/DS00Spx1CV902MCtPj5WknGlR102V5HFkDe.m3u8">
+            <track label="thumbnails" default kind="metadata" src="https://image.mux.com/DS00Spx1CV902MCtPj5WknGlR102V5HFkDe/storyboard.vtt" />
```
Comment (luwes) [inline examples/media-elements/media-playlist.html:28]: this is really cool! `type="hls-video"` nice!

**Comment by luwes** [comment]
I'm more leaning to keeping the media elements (source) out of the Media Chrome repo. Examples in, for sure.

[`media-playlist`](https://github.com/muxinc/media-playlist) is not really a video engine either, more like a source / media selector
[`responsive-source`](https://github.com/muxinc/responsive-source) would also fall in this category 

Since it's not UI my s1 is to not include it but I'm not 100% sure what the extras can include.

--- 

If it's for reducing friction w/ releasing etc, there's still some things we could do for the standalone pkgs like auto releasing on PR merges etc.

**Comment by heff** [comment]
> If we add those, do they go in that project? Or here?

Here (media-chrome), because media-playlist won't be the only media element with playlist functionality. e.g. [Youtube](https://developers.google.com/youtube/iframe_api_reference) with `player.nextVideo()`. That makes it a [media-ui-extensions](https://github.com/video-dev/media-ui-extensions/issues) detail to solve, but otherwise we've been pretty clear and consistent with the line between "chrome" and "media". While I feel like this is a _special_ or _supportive_ media element, it also feels like a slippery slope introducing media elements into media chrome. But the discoverability of the whole ecosystem, including other media elements, is something that's important to solve well.

**Comment by heff** [comment]
> Must be a custom element loading issue. I'll dig into when I get a minute.

Confirmed, the load order of media-playlist vs media-chrome scripts matters for some reason. Not clear why yet. If media playlist is loaded _before_ media chrome, the problem happens. That feels backwards from what I'd expect.

**Comment by luwes** [comment]
could it be before this line `document.createElement(playlistItem.getAttribute('type'));` if it's a custom element we need to wait until it's defined? `customElements.whenDefined()` 

or possibly upgrade it right after so the attributeCallback will be triggered when setting the attributes  
https://developer.mozilla.org/en-US/docs/Web/API/CustomElementRegistry/upgrade

also make sure you attach the events before setting the attributes, though less important, `loadstart` will be delayed by a tick when setting the src.

**Comment by luwes** [comment]
I confirmed this is caused by https://github.com/muxinc/hls-video-element
if you add in my fork `<script type="module" src="https://unpkg.com/@luwes/hls-video-element@0"></script>` it works

maybe temporarily use just plain video and audio tags to get this example in?

(https://github.com/muxinc/media-playlist has a small bug where `media.muted` doesn't seem to work)

**Review by luwes — APPROVED** [review]
LGTM, small hls-video fix needed but don't want to block this from merging

**Pull request merged**